openwisp_monitoring/check/classes/iperf3.py

```diff
@@ -201,6 +202,14 @@ def check(self, store=True):
                 )
             )
             return
+        device_monitoring = DeviceMonitoring.objects.filter(
+            device=self.related_object
+        ).first()
```

Comment:
don't we have the device object somewhere already available here?

Reply:
yes, `DeviceMonitoring` is OneToOne related with `Device` :+1: